docs/manual/howto/git.rst

```diff
@@ -0,0 +1,104 @@
+.. _using-git:
+
+=============
+Using ``git``
+=============
+
+``git`` is the version control system that is used to manage all of the source
+code. It is very powerful, but might be frightening at first.
+This page should give you a quick overview, but for a complete guide you will
+have to search the web on your own.
+Another great resource to get started practically without having to try out the
+newly-learned commands on a pre-existing repository is
+`learn git branching <https://learngitbranching.js.org>`_.
+You should probably learn the basic ``git`` vocabulary and then come back to
+find out how you can use all that practically. This guide will be oriented on
+how to create a pull request and things might be in a different order compared
+to the introductory guides.
+
+.. warning:: This guide is not complete and never will be. If something isn't
+   clear, consult other sources until you are confident you know what you are
+   doing.
+
+I want to try out a feature somebody is working on
+==================================================
+If you see a pull request on `GitHub <https://www.github.com/qtile/qtile/pulls>`_
+that you want to try out, have a look at the line where it says::
+
+  user wants to merge n commits into qtile:master from user:branch
+
+Right now you probably have one *remote* from which you can fetch changes, the
+``origin``. If you cloned ``qtile/qtile``, ``git remote show origin`` will spit
+out the *upstream* url. If you cloned your fork, ``origin`` points to it and you
+probably want to ``git remote add upstream https://www.github.com/qtile/qtile``.
+To try out somebody's work, you can add his fork as a new remote::
+
+  git remote add user https://www.github.com/user/qtile
```

Comment:
Interesting. I don't do it this way, I do:
```
git fetch upstream pull/<id>/head:<new-branch-name>
git checkout <new-branch-name>
```
That way I don't add additional remotes.

Reply:
Oooh! This maybe seems nicer... perhaps we should describe this approach too? Not sure I understand how it works actually. If the upstream is to full qtile/qtile URL, how does the ID get resolved? Is this GitHub being clever, or does this work with many/all git servers?

Reply:
So, yes, `upstream` is full qtile/qtile. ID doesn't get resolved, you enter it. So, to test this PR I did:
```
git fetch upstream pull/2333/head:pr2333
git checkout pr2333
cd docs
make html
```

Reply:
Huh, well this is way better. @ep12 what do you think about using this instead for the docs?

Reply:
Haven't used it, but sure if is better then this should be changed of course! Thanks for the hint btw. 😁